Registration/RegisterWithAzure.psm1

```diff
@@ -9,6 +9,45 @@ You must also have access to an account / directory that is an owner or contribu
 
 #>
 
+function Initialize-AzEnvironment{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$false)]
+        [String] $Name = 'CustomCloud',
+
+        [Parameter(Mandatory=$true, ParameterSetName='CloudManifest')]
+        [String] $CloudManifestFilePath,
+
+        [Parameter(Mandatory=$true, ParameterSetName='CloudARMEndpoint')]
+        [String] $CloudARMEndpoint
+    )
+
+    if ( $PSCmdlet.ParameterSetName -eq 'CloudManifest' ){
+        $cloudJson = Get-Content $CloudManifestFilePath -Raw | ConvertFrom-Json
+        $cloudJsonContent = $cloudJson.DeploymentData
+        $CloudARMEndpoint = $cloudJsonContent.CustomEnvironmentEndpoints.CustomCloudARMEndpoint
+    }
+
+    $fullUri = $CloudARMEndpoint.TrimEnd('/')+"/metadata/endpoints?api-version=2015-01-01"
+    $response = Invoke-RestMethod -Uri $fullUri -ErrorAction Stop -UseBasicParsing -TimeoutSec 30 -Verbose
+    Write-Verbose -Message "Endpoints: $(ConvertTo-Json $response)" -Verbose
+    $endpoints = @{
```

Comment:
Add-AzEnvironment takes -ArmEndpoint as a parameter which would do the metadata querying by default. You dont need to query the metadata endpoint to get the values